Content.Server/Roles/NinjaRole.cs

```diff
@@ -0,0 +1,41 @@
+using Content.Shared.Roles;
+
+namespace Content.Server.Roles;
+
+/// <summary>
+/// Stores the ninja's objectives in the mind so if they die the rest of the greentext persists.
+/// </summary>
+public sealed class NinjaRole : TraitorRole
+{
+    public NinjaRole(Mind.Mind mind, AntagPrototype antagPrototype) : base(mind, antagPrototype) { }
+
+    /// <summary>
+    /// Number of doors that have been doorjacked, used for objective
+    /// </summary>
+    [ViewVariables]
+    public int DoorsJacked = 0;
+
+    /// <summary>
+    /// Research nodes that have been downloaded, used for objective
+    /// </summary>
+    [ViewVariables]
+    public HashSet<string> DownloadedNodes = new();
+
+    /// <summary>
+    /// Warp point that the spider charge has to target
+    /// </summary>
+    [ViewVariables]
+    public EntityUid? SpiderChargeTarget = null;
+
+    /// <summary>
+    /// Whether the spider charge has been detonated on the target, used for objective
+    /// </summary>
+    [ViewVariables]
+    public bool SpiderChargeDetonated;
+
+    /// <summary>
+    /// Whether the comms console has been hacked, used for objective
+    /// </summary>
+    [ViewVariables]
+    public bool CalledInThreat;
```

Comment:
shouldnt this all be in the component

Reply:
the component is destroyed when dying (microbomb) so it wont show in end round info

Reply:
Should roles be datadefs coz none of this is saved on save-load.

Reply:
tbf ninja is only thing that stores data in the role so maybe maybe not

Reply:
role is datadef now i think